### PR TITLE
Feature/25 baseurl

### LINKED ISSUE #25: Configure Axios BaseUrls

### LINKED ISSUE #39: Move api base url to env file

### PR DESCRIPTION
**Pull request by prekel**
closes #25 closes #39 started #32 touched #46

### CODE REVIEW

File: .env.production [hunk 1]
```diff
@@ -0,0 +1 @@
+VITE_VNDBRE_PROXY_BASEURL="https://vndbre-proxy.azurewebsites.net/api/v1/"
```
Comment (moetazaneta) [inline .env.production:1]: I think VNDBRE prefix is excessive, maybe just name it VITE_API_BASE_URL

File: package.json [hunk 2]
```diff
@@ -3,26 +3,28 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@bbob/preset-react": "^2.7.1",
+    "@bbob/react": "^2.7.1",
```
Comment (moetazaneta) [inline package.json]: bbob shouldn't be there

File: src/api/index.ts [hunk 3]
```diff
@@ -16,5 +16,5 @@ export const queryClient = new QueryClient({
  * TODO: CHANGE BASE URL.
  */
 export const http: AxiosInstance = axios.create({
-  baseURL: 'https://vndbre-proxy.azurewebsites.net/api/v1/',
+  baseURL: import.meta.env.VITE_VNDBRE_PROXY_BASEURL as string,
```
Comment (moetazaneta) [inline src/api/index.ts]: There is a better way to add typings https://vitejs.dev/guide/env-and-mode.html#intellisense-for-typescript

File: .env.development [hunk 4]
```diff
@@ -0,0 +1 @@
+VITE_VNDBRE_PROXY_BASEURL="http://localhost:7879/api/v1/"
```
Comment (moetazaneta) [inline .env.development:1]: Developers don't have backend locally running

**Review by moetazaneta — CHANGES_REQUESTED** [review]

File: .env.development (hunk 4 above)
Comment (moetazaneta) [reply .env.development:1]: I mean front-end developers

File: package.json [hunk 5]
```diff
@@ -18,8 +18,9 @@
     "react-router-dom": "^6.0.0-beta.3"
   },
   "scripts": {
-    "dev": "vite",
-    "build": "tsc && vite build",
+    "dev": "vite --mode production",
+    "dev-dev": "vite --mode development",
```
Comment (moetazaneta) [inline package.json]: I suggest rename to `dev` and `dev:production`

**Review by NightOfNights — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -6,7 +6,7 @@ This project was bootstrapped with [Create React App](https://github.com/faceboo
 
 In the project directory, you can run:
 
-### `npm start`
+### `npm dev`
```
Comment (NightOfNights) [inline README.md]: `npm run dev`

**Review by moetazaneta — CHANGES_REQUESTED** [review]
I despise `dev-dev`.

**Review by moetazaneta — APPROVED** [review]

**Pull request merged**